src/lib/piwik.js

```diff
@@ -0,0 +1,62 @@
+import config, { getPiwik, getDimensions } from '../config'
+import { getMethod } from '../helpers'
+
+let intr
+let coll = []
+
+export default function piwik (method, ...args) {
+  
+  const {piwikEndpoint, piwikSiteId, userId} = getPiwik();
+  const client = window.Piwik;
+  const tracker = client.getTracker(piwikEndpoint, piwikSiteId);
+
+  if ( userId != null ){
+    tracker.setUserId(userId);
+  }
+
+  if(method === 'pageview'){
+    // Page view events, send dimension
+    const dimensions = getDimensions();
+
+    let url = window.location.href;
+    if(args.length > 0 & args[0]){
+      tracker.setCustomUrl(args[0]);
+    }
+
+    if(args.length > 1 & args[1]){
+      tracker.setDocumentTitle(args[1]);
+    }
+
+    if (dimensions){
+      Object.entries(dimensions).map(row => {
+        const [key, value] = row;
+        const newKey = parseInt(key.match('mension([0-9]+)')[1]);
```

Comment:
`key.match` may return null (if no match is found)
And `null[1]` will throw error. Will we need to handle this potential error case?